test/stablex/batch_exchange.js

```diff
@@ -221,7 +221,7 @@ contract("BatchExchange", async accounts => {
       const batchExchange = await setupGenericStableX()
 
       const id = await batchExchange.placeOrder.call(0, 1, 3, 10, 20, { from: user_1 })
-      const currentStateIndex = (await batchExchange.getCurrentBatchId()).toNumber()
+      const currentStateIndex = (await batchExchange.getCurrentBatchId.call()).toNumber()
```

Comment:
is this really the only place where we weren't already using `.call`? Is it possible that there are other flakey tests with exactly this problem?

Reply:
Does this really change something? I tried the following statement in the `truffle console` and it doesn't lead to anything being mined on my local ganache.

```js
let batchExchange = await BatchExchange.deployed()
await batchExchange.getCurrentBatchId()
```